internal/deb/extract.go

```diff
@@ -277,6 +282,140 @@ func extractData(dataReader io.Reader, options *ExtractOptions) error {
 	return nil
 }
 
+type pendingHardLink struct {
+	path         string
+	extractInfos []ExtractInfo
+}
+
+type extractHardLinkOptions struct {
+	*ExtractOptions
+	pendingLinks map[string][]pendingHardLink
+}
+
+// extractHardLinks iterates through the tarball a second time to extract the
+// hard links that were not extracted in the first pass.
+func extractHardLinks(pkgReader io.ReadSeeker, opts *extractHardLinkOptions) error {
+	offset, err := pkgReader.Seek(0, io.SeekStart)
```

Comment:
We discussed this in the call last week: the seeking should be done by whoever calls extractHardLinks, instead of expecting this function to know it's going to be called at a point it's not supposed to read.  Note how extractData itself doesn't seek.

Reply:
Agree, I forgot to publish the comments that we made during the meeting. Will address it now, thanks!

Reply:
Done in https://github.com/canonical/chisel/pull/179/commits/d6ea9947fd5e5128cf326a08b7de2919c074ec60.